### ISSUE TITLE
no-setup-in-describe triggers on new before/beforeEach/after/afterEach.timeout

### ISSUE THREAD

**Issue by perrin4869** [lo1tuma/eslint-plugin-mocha#364]
This was added here: https://github.com/mochajs/mocha/pull/5231
And the typings were just merged: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71149

```
describe("my suite", () => {
  before(() => {}).timeout(1000); // triggers mocha/no-setup-in-describe
});
```

The above should now be valid (on mocha 11), but is triggering
